sky/backends/cloud_vm_ray_backend.py

```diff
@@ -278,6 +279,13 @@ def check_ip():
                 print('INFO: Reserved IPs:', ip_list)
                 ip_list_str = '\\n'.join(ip_list)
                 sky_env_vars_dict['SKY_NODE_IPS'] = ip_list_str
+
+                if {cluster_ips_sorted!r}:
+                    cluster_ips_map = {{ip: i for i, ip in enumerate({cluster_ips_sorted!r})}}
+                    ip_rank_list = sorted(ip_list, key=cluster_ips_map.get)
+                    ip_rank_map = {{ip: i for i, ip in enumerate(ip_rank_list)}}
```

Comment:
nit: Add a comment for this block or add a docstr for the `add_gang_scheduling_placement_group` function to indicate the purpose of `cluster_ips_sorted`, e.g. cluster_ips_sorted is used to ensure the `SKY_NODE_RANK` assigned later will be in a deterministic order.